### PR TITLE
Add contributors bot that is triggered every day to update contributors-record.md in afterword

### PR DESCRIPTION
**Pull request by malvikasharan**
### Summary

Fixes #1330 

* Created a GitHub action that updates the contributors record file every day (currently in test mode-> update to 23 hr)
* Tested it on my fork and it works, locally I had many errors in building the book so had to fix some small errors as well
* Updated 'contributors.md' file with the comments in the issue #1584 

### List of changes proposed in this PR (pull-request)

* Created a GitHub action that updates the contributors record file every day (currently in test mode-> update to 23 hr)
* Tested it on my fork and it works, locally I had many errors in building the book so had to fix some small errors as well
* Updated 'contributors.md' file with the comments in the issue #1584 

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?
- [ ] GitHub Action works as expected.

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by malvikasharan** [comment]
@sgibson91 Can you please test this when you have time to see if the yml file makes sense and that it works for you too.

File: .github/workflows/contributors-record.yml
Comment (malvikasharan) [inline .github/workflows/contributors-record.yml:7]:
```suggestion
    - cron:  '* */23 * * *'
```

**Review by malvikasharan — COMMENTED** [review]
Update this section before merging to change the schedule from 1min -> 23 hours.

File: .github/workflows/contributors-record.yml
Comment (KirstieJane) [reply .github/workflows/contributors-record.yml:7]: OMG - you know me - I don't really read most of my GitHub notifications but this one made me laugh!! Good catch! When do you think we'll hit the event horizon and need to update the contributors page every minute 😉

**Comment by KirstieJane** [comment]
Do you need to pull out the unrelated files in this PR @malvikasharan?

I love the work! (This isn't a proper review because I'm not really qualified - but great work!)

**Comment by malvikasharan** [comment]
Thank you @KirstieJane 💐 I had issues building the book locally so updated some files - but I can definitely leave out what's not needed for this action specifically.

File: .github/workflows/contributors-record.yml
Comment (malvikasharan) [reply .github/workflows/contributors-record.yml]: I can see in my notification how 1 min update can really bother people who have all notification on 😅

File: .github/workflows/contributors-record.yml
Comment (martinagvilas) [inline .github/workflows/contributors-record.yml:7]: I might be super wrong, but I think this set up will trigger the action every minute past 23 hours. I would use instead the following: `'0 23 * * *'` to trigger the action only once at 23 pm. I like using [this page](https://crontab.guru/) to check how I write my cron jobs, in case anyone reading find that useful.

File: tests/contributors.sh
Comment (martinagvilas) [inline tests/contributors.sh:13]:
```suggestion
# Get linenumber where all contributors list starts
```

**Review by martinagvilas — COMMENTED** [review]
This looks amazing! 

Just one comment: Are we sure this action will successfully trigger a PR? I might be wrong, but I think the action is adding origin as the master branch of the main repo (when this gets merged, not right now), and then is asking to create a PR to master as well. I might be missing something though, it's been a while since I've created a github action 😆

File: .github/workflows/contributors-record.yml
Comment (malvikasharan) [reply .github/workflows/contributors-record.yml:7]: Ah, yes. Thank you for catching that. I need them to run once per day (even less would be ok too - like once in 3 days). That cronguru link is pretty great. Thank you for sharing - so with the current change '30 09 * * *' I can expect this to run at 9:30 everyday I hope.

**Comment by malvikasharan** [comment]
@martinagvilas Thank you for the review. This action at the moment is not sending a PR, but triggering a push to my fork. I don't know how it will be after I merge this. I will have a mother look at if there is a better way to send PR through GitHub action other than what I am using.

**Comment by martinagvilas** [comment]
> @martinagvilas Thank you for the review. This action at the moment is not sending a PR, but triggering a push to my fork. I don't know how it will be after I merge this. I will have another look maybe there is a better way to send PR through GitHub action other than what I am using.

@malvikasharan makes sense 👍  If it doesn't work when merged, one thing to try might be to do the checkout on a branch that is not master by setting `ref` to that branch when using [this action](https://github.com/actions/checkout#usage).

**Review by malvikasharan — COMMENTED** [review]
Will merge now to test if the bot works.

**Comment by martinagvilas** [comment]
@malvikasharan I think it might be working now, you can see the PR in your fork 🚀  Since the fork was not merged, each time the action is run the changes appear as a new commit in the open PR. I will double check tomorrow 9:30 (scheduled time) to see if it works as expected.

**Comment by malvikasharan** [comment]
Thank you @martinagvilas. This seems to work now 💃 Thank you for fixing the issue! ✨ @sgibson91 Can you please have a quick look and approve this?

**Review by sgibson91 — APPROVED** [review]

**Comment by malvikasharan** [comment]
Thank you @sgibson91 🌺🤞

File: .github/workflows/contributors-record.yml
Comment (malvikasharan) [inline .github/workflows/contributors-record.yml:7]:
```suggestion
    - cron: '00 13 * * *'
```

**Pull request merged**
